code/datums/mob_stagehandler.dm

```diff
@@ -0,0 +1,44 @@
+//Meant to handle stages of transitions or effects on mobs instead of humans having like 8 vars defined just for SCPs
+/datum/stageHandler
+	var/list/stages
+
+/datum/stageHandler/New()
+	. = ..()
+	LAZYINITLIST(stages)
+
+///Creates a new stage to track
+/datum/stageHandler/proc/createStage(stageID)
+	if(stages[stageID])
+		return 0
+	stages[stageID] = 0
+	return 1
+
+///Returns value of a stage
+/datum/stageHandler/proc/getStage(stageID)
+	if(!stages[stageID])
```

Comment:
This also checks for 0, not null, so after creation it gets set to 0 but gets treated as not existing